### PR TITLE
Assets and Sample App

### PR DESCRIPTION
**Pull request by chrisanderberg**
This commit contains all the assets and a sample app for integrating Tealium into a React Native application.

### CODE REVIEW

File: android/TealiumModule.java [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+
+    }
+
+    @Override
+    public String getName() {
+        return "TealiumModule";
+    }
+
+    @ReactMethod
+    public void show(String message) {
```
Comment (tamayok) [inline android/TealiumModule.java]: Remove this too.

File: TealiumSampleApp/ios/TealiumModule.h [hunk 2]
```diff
@@ -0,0 +1,17 @@
+//
+//  TealiumRNModule.h
+//  do-over
```
Comment (jonwong-tealium) [inline TealiumSampleApp/ios/TealiumModule.h]: remove `do-over`

File: TealiumSampleApp/ios/TealiumModule.h [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+//
+//  Created by Christopher Anderberg on 1/8/18.
+//  Copyright © 2018 650 Industries, Inc. All rights reserved.
```
Comment (jonwong-tealium) [inline TealiumSampleApp/ios/TealiumModule.h]: Change to Tealium or remove.

File: TealiumSampleApp/ios/TealiumModule.m [hunk 4]
```diff
@@ -0,0 +1,118 @@
+//
+//  TealiumRNModule.m
+//  DoOver
+//
+//  Created by Christopher Anderberg on 1/8/18.
+//  Copyright © 2018 650 Industries, Inc. All rights reserved.
```
Comment (jonwong-tealium) [inline TealiumSampleApp/ios/TealiumModule.m]: Same comment as above.  Remove DoOver and 650 Industries, Inc.

File: TealiumSampleApp/ios/TealiumSampleApp-tvOS/Info.plist [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+		<string>UIInterfaceOrientationPortrait</string>
+		<string>UIInterfaceOrientationLandscapeLeft</string>
+		<string>UIInterfaceOrientationLandscapeRight</string>
+	</array>
+	<key>UIViewControllerBasedStatusBarAppearance</key>
+	<false/>
+	<key>NSLocationWhenInUseUsageDescription</key>
+	<string></string>
+	<key>NSAppTransportSecurity</key>
+	<!--See http://ste.vn/2015/06/10/configuring-app-transport-security-ios-9-osx-10-11/ -->
```
Comment (jonwong-tealium) [inline TealiumSampleApp/ios/TealiumSampleApp-tvOS/Info.plist]: Was this comment supposed to be left in?

File: TealiumSampleApp/ios/TealiumSampleApp/AppDelegate.h [hunk 6]
```diff
@@ -0,0 +1,16 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
```
Comment (jonwong-tealium) [inline TealiumSampleApp/ios/TealiumSampleApp/AppDelegate.h]: This is Tealium source code, right?  If so, update or remove the header.

File: TealiumSampleApp/ios/TealiumSampleApp/AppDelegate.m [hunk 7]
```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
```
Comment (jonwong-tealium) [inline TealiumSampleApp/ios/TealiumSampleApp/AppDelegate.m]: Same comment as above.  If so, update or remove the header.

File: TealiumSampleApp/ios/TealiumSampleApp/Info.plist [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+		<string>UIInterfaceOrientationPortrait</string>
+		<string>UIInterfaceOrientationLandscapeLeft</string>
+		<string>UIInterfaceOrientationLandscapeRight</string>
+	</array>
+	<key>UIViewControllerBasedStatusBarAppearance</key>
+	<false/>
+	<key>NSLocationWhenInUseUsageDescription</key>
+	<string></string>
+	<key>NSAppTransportSecurity</key>
+	<!--See http://ste.vn/2015/06/10/configuring-app-transport-security-ios-9-osx-10-11/ -->
```
Comment (jonwong-tealium) [inline TealiumSampleApp/ios/TealiumSampleApp/Info.plist]: Should this comment be left in?

File: TealiumSampleApp/ios/TealiumSampleApp/main.m [hunk 9]
```diff
@@ -0,0 +1,18 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
```
Comment (jonwong-tealium) [inline TealiumSampleApp/ios/TealiumSampleApp/main.m]: Same comment as above.  Remove or update.

File: TealiumSampleApp/ios/TealiumSampleAppTests/TealiumSampleAppTests.m [hunk 10]
```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
```
Comment (jonwong-tealium) [inline TealiumSampleApp/ios/TealiumSampleAppTests/TealiumSampleAppTests.m]: Same comment as above.  Update or remove.

File: TealiumSampleApp/ios/TealiumSampleAppTests/TealiumSampleAppTests.m [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+
+#import <UIKit/UIKit.h>
+#import <XCTest/XCTest.h>
+
+#import <React/RCTLog.h>
+#import <React/RCTRootView.h>
+
+#define TIMEOUT_SECONDS 600
```
Comment (jonwong-tealium) [inline TealiumSampleApp/ios/TealiumSampleAppTests/TealiumSampleAppTests.m]: Prefer using const variables over preprocessor macros.

File: TealiumSampleApp/ios/TealiumSampleAppTests/TealiumSampleAppTests.m [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+#define TEXT_TO_LOOK_FOR @"Welcome to React Native!"
```
Comment (jonwong-tealium) [inline TealiumSampleApp/ios/TealiumSampleAppTests/TealiumSampleAppTests.m]: Same comment as above.

File: TealiumSampleApp/ios/TealiumSampleAppTests/TealiumSampleAppTests.m [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+    if ([self findSubviewInView:subview matching:test]) {
+      return YES;
+    }
+  }
+  return NO;
+}
+
+- (void)testRendersWelcomeScreen
+{
+  UIViewController *vc = [[[RCTSharedApplication() delegate] window] rootViewController];
```
Comment (jonwong-tealium) [inline TealiumSampleApp/ios/TealiumSampleAppTests/TealiumSampleAppTests.m]: Minor: Objective-C prefers using full names.  Update `vc` to `viewController`.

File: android/TealiumModule.java [hunk 14]
```diff
@@ -0,0 +1,316 @@
...
+    private static ReactApplicationContext mReactApplicationContext;
+
+    private static String mTealiumInstanceName;
+    private static boolean mIsLifecycleAutotracking = false;
+    private boolean mDidTrackInitialLaunch = false;
+
+    public TealiumModule(ReactApplicationContext reactContext) {
+        super(reactContext);
+        mReactApplicationContext = reactContext;
+
```
Comment (jonwong-tealium) [inline android/TealiumModule.java]: minor: remove newline.

File: android/TealiumModule.java [hunk 15]
```diff
@@ -0,0 +1,316 @@
...
+                default:
+                    throw new IllegalArgumentException("Could not set volatile data for key: " + key);
+            }
+        }
+    }
+
+    @ReactMethod
+    public void setPersistentData(ReadableMap data) {
+        final Tealium instance = Tealium.getInstance(mTealiumInstanceName);
+        if (instance == null) {
```
Comment (jonwong-tealium) [inline android/TealiumModule.java]: minor: add space after `final Tealium instance` line to match `setVolatileData` or update `setVolatileData` formatting.

File: android/TealiumModule.java [hunk 16]
```diff
@@ -0,0 +1,316 @@
...
+                default:
+                    Log.e(BuildConfig.TAG, "Invalid key type. Use array of strings");
+            }
+        }
+    }
+
+    @ReactMethod
+    public void getVolatileData(String key, Callback callback) {
+        final Tealium instance = Tealium.getInstance(mTealiumInstanceName);
+        if (instance == null) {
```
Comment (jonwong-tealium) [inline android/TealiumModule.java]: same comment as above regarding formatting.

File: android/TealiumModule.java [hunk 17]
```diff
@@ -0,0 +1,316 @@
...
+            Log.e(BuildConfig.TAG, "Attempt to get volatile data, but Tealium not enabled for instance name: " + mTealiumInstanceName);
+            return;
+        }
+
+        callback.invoke(instance.getDataSources().getVolatileDataSources().get(key));
+    }
+
+    @ReactMethod
+    public void getPersistentData(String key, Callback callback) {
+        final Tealium instance = Tealium.getInstance(mTealiumInstanceName);
```
Comment (jonwong-tealium) [inline android/TealiumModule.java]: same comment as above regarding formatting.

File: android/TealiumPackage.java [hunk 18]
```diff
@@ -0,0 +1,30 @@
...
+    public List<NativeModule> createNativeModules(ReactApplicationContext reactContext) {
+        List<NativeModule> modules = new ArrayList<>();
+
+        modules.add(new TealiumModule(reactContext));
+        return modules;
+    }
+
+    @Override
+    public List<ViewManager> createViewManagers(ReactApplicationContext reactContext) {
+//        return Collections.emptyList();
```
Comment (jonwong-tealium) [inline android/TealiumPackage.java]: remove if not needed.

File: ios/TealiumModule.h [hunk 19]
```diff
@@ -0,0 +1,17 @@
+//
+//  TealiumRNModule.h
+//  do-over
```
Comment (jonwong-tealium) [inline ios/TealiumModule.h]: remove `do-over`.

File: ios/TealiumModule.h [hunk 20]
```diff
@@ -0,0 +1,17 @@
...
+//
+//  Created by Christopher Anderberg on 1/8/18.
+//  Copyright © 2018 650 Industries, Inc. All rights reserved.
```
Comment (jonwong-tealium) [inline ios/TealiumModule.h]: Update or remove 650 Industries, Inc.

File: ios/TealiumModule.m [hunk 21]
```diff
@@ -0,0 +1,118 @@
+//
+//  TealiumRNModule.m
+//  DoOver
```
Comment (jonwong-tealium) [inline ios/TealiumModule.m]: remove `DoOver`

File: ios/TealiumModule.m [hunk 22]
```diff
@@ -0,0 +1,118 @@
...
+//
+//  Created by Christopher Anderberg on 1/8/18.
+//  Copyright © 2018 650 Industries, Inc. All rights reserved.
```
Comment (jonwong-tealium) [inline ios/TealiumModule.m]: Update or remove 650 Industries, Inc.

**Pull request merged**